Sming/SmingCore/Platform/System.cpp

```diff
@@ -30,11 +32,20 @@ void SystemClass::initialize()
 	state = eSS_Intializing;
 
 	// Initialise the global task queue
-	system_os_task(__taskHandler, USER_TASK_PRIO_1, taskQueue, TASK_QUEUE_LENGTH);
+	system_os_task(taskHandler, USER_TASK_PRIO_1, taskQueue, TASK_QUEUE_LENGTH);
 
 	system_init_done_cb(staticReadyHandler);
 }
 
+bool SystemClass::queueCallback(TaskCallback callback, os_param_t param)
```

Comment:
Maybe you can add priority - a third optional parameter to queueCallback with default value USER_TASK_PRIO_1 ?

Reply:
If I do that, I'd need to check the parameter and fail it. I could create the second task queue on-the-fly, but should also include some way of checking we're not in interrupt context. Alternatively I could add another method to SystemClass to create the elevated task queue. Or I could include the code to create the second task queue using `#define`. It's getting a bit complicated, which is why I'm thinking just leave it until the need arises? It wouldn't affect any existing code...

Reply:
> which is why I'm thinking just leave it until the need arises?

Ok, that's fine.